src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
     /**
```

Comment:
Don't begin interface or method implementation with an empty line. It is unnecessary and does not follow Java code style conventions.